server/util/redact/redact_test.go

```diff
@@ -0,0 +1,165 @@
+package redact_test
+
+import (
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/proto/build_event_stream"
+	"github.com/buildbuddy-io/buildbuddy/proto/command_line"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/buildbuddy-io/buildbuddy/server/util/redact"
+	"github.com/stretchr/testify/assert"
+)
+
+func singleFile() *build_event_stream.File {
+	return &build_event_stream.File{
+		Name: "afile",
+		File: &build_event_stream.File_Uri{
+			Uri: "213wZJyTUyhXkj381312@uri",
+		},
+	}
+}
+
+func singleFiles() []*build_event_stream.File {
+	return []*build_event_stream.File{
+		singleFile(),
+	}
+}
+
+func TestRedact(t *testing.T) {
+	redactor := redact.NewStreamingRedactor(testenv.GetTestEnv(t))
+
+	buildStarted := &build_event_stream.BuildStarted{
+		OptionsDescription: "213wZJyTUyhXkj381312@foo",
+	}
+	redactor.RedactMetadata(&build_event_stream.BuildEvent{
+		Payload: &build_event_stream.BuildEvent_Started{Started: buildStarted},
+	})
+
+	assert.Equal(t, "foo", buildStarted.OptionsDescription)
+
+	shellOption := &command_line.Option{
+		CombinedForm: "--client_env=SHELL=/bin/bash",
+		OptionName:   "client_env",
+		OptionValue:  "SHELL=/bin/bash",
+	}
+	secretOption := &command_line.Option{
+		CombinedForm: "--client_env=SECRET=codez",
+		OptionName:   "client_env",
+		OptionValue:  "SECRET=codez",
+	}
+	structuredCommandLine := &command_line.CommandLine{
+		CommandLineLabel: "label",
+		Sections: []*command_line.CommandLineSection{
+			{
+				SectionLabel: "command",
+				SectionType: &command_line.CommandLineSection_OptionList{
+					OptionList: &command_line.OptionList{
+						Option: []*command_line.Option{
+							shellOption,
+							secretOption,
+						},
+					},
+				},
+			},
+		},
+	}
+	redactor.RedactMetadata(&build_event_stream.BuildEvent{
+		Payload: &build_event_stream.BuildEvent_StructuredCommandLine{StructuredCommandLine: structuredCommandLine},
+	})
+
+	// TODO(bduffs): Migrate env redaction to redactor and enable these assertions.
+	// assert.Equal(t, "--client_env=SECRET=<REDACTED>", secretOption.OptionValue)
+	// assert.Equal(t, "secret=<REDACTED>", secretOption.OptionValue)
+
+	optionsParsed := &build_event_stream.OptionsParsed{
+		CmdLine:         []string{"213wZJyTUyhXkj381312@foo"},
+		ExplicitCmdLine: []string{"213wZJyTUyhXkj381312@explicit"},
+	}
+	redactor.RedactMetadata(&build_event_stream.BuildEvent{
+		Payload: &build_event_stream.BuildEvent_OptionsParsed{OptionsParsed: optionsParsed},
+	})
+
+	assert.Equal(t, []string{"foo"}, optionsParsed.CmdLine)
+	assert.Equal(t, []string{"explicit"}, optionsParsed.ExplicitCmdLine)
+
+	actionExecuted := &build_event_stream.ActionExecuted{
+		Stdout:             singleFile(),
```

Comment:
What do you think about passing the URI as an argument to singleFile() so that person reading the test can see both the input and the expected output in the test?

Reply:
SG, done (and renamed `singleFile` to `fileWithURI`)